lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -258,6 +259,23 @@ def configure_app_extension_api_only_for_target(target)
             config.build_settings['APPLICATION_EXTENSION_API_ONLY'] = 'YES'
           end
         end
+
+        def user_swift_version
+          declared_swift_versions = analysis_result
+            .targets
+            .map(&:user_project)
+            .compact
+            .flat_map(&:native_targets)
+            .flat_map(&:build_configurations)
+            .flat_map(&:build_settings)
+            .map { |settings| settings['SWIFT_VERSION'] }
+            .compact
+            .uniq
+
+          # TODO: Assert only 1 or 0 unique Swift Versions declared
+
+          declared_swift_versions.first
```

Comment:
No, it's not possible, LLVM does not allow that. 
It's possible that an app uses 3.0 and extension could use 2.3 - they would be in the same xcodeproject. Which is a use case I was saying might be worth being aware of.